pkg/tag/api/api.go

```diff
@@ -121,8 +121,42 @@ func (s *TagService) CreateTag(
 		)
 		return nil, s.reportInternalServerError(ctx, err, req.EnvironmentId, localizer)
 	}
-	// Save in the DB
-	if err := s.tagStorage.UpsertTag(ctx, tag); err != nil {
+
+	var event *eventproto.Event
+	var actualTag *domain.Tag
+	err = s.mysqlClient.RunInTransactionV2(ctx, func(ctxWithTx context.Context, _ mysql.Transaction) error {
+		if err := s.tagStorage.UpsertTag(ctxWithTx, tag); err != nil {
+			return err
+		}
+		// Fetch the actual tag from DB to get correct final state after upsert
+		actualTag, err = s.tagStorage.GetTagByName(ctxWithTx, tag.Name, tag.EnvironmentId, tag.EntityType)
```

Comment:
When updating an existing tag, we must use the actual tag info instead of using the tag generated by the `domain.NewTag` because it will report an incorrect ID that doesn't exist and a different `created_at` value.